client/ayon_core/modules/deadline/plugins/publish/collect_user_credentials.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+"""Collect user credentials
+
+Requires:
+    context -> project_settings
+
+Provides:
+    context -> deadline_require_authentication (bool)
+    context -> deadline_auth (tuple (str, str)) - (username, password) or None
+"""
+import pyblish.api
+
+
+class CollectUserCredentials(pyblish.api.ContextPlugin):
```

Comment:
Can we include `Deadline` in the class name to avoid any potential conflict with other plug-ins in the future.

```suggestion
class CollectDeadlineUserCredentials(pyblish.api.ContextPlugin):
```